src/libstd/sys/unix/fd.rs

```diff
@@ -28,7 +33,8 @@ const READ_LIMIT: usize = libc::ssize_t::MAX as usize;
 
 impl FileDesc {
     pub fn new(fd: c_int) -> FileDesc {
-        FileDesc { fd }
+        assert_ne!(fd, -1);
```

Comment:
Would it make sense to add the same assertion in `FromRawFd`? If this is deemed too expensive it could be a debug assertion.

Reply:
Oh, this function probably backs `FromRawFd`... never mind.

Reply:
This function is safe, and it contains an `unsafe` block, so it ought to be responsible for maintaining the invariant.